compute/cker/include/cker/train/operation/Pad.h

```diff
@@ -0,0 +1,168 @@
+        const auto out_offset = h * out_width;
+        // copy a row of input data to output data
+        std::memcpy(output_data + out_offset, input_data + in_offset, out_width * sizeof(T));
+      }
+      break;
+    }
+    case 3: // HWC
+    {
+      const int32_t out_depth = output_shape.Dims(0);
+      const int32_t out_height = output_shape.Dims(1);
+      const int32_t out_width = output_shape.Dims(2);
+      const int32_t out_plain_size = out_height * out_width;
+      const int32_t in_height = input_shape.Dims(1);
+      const int32_t in_width = input_shape.Dims(2);
+      const int32_t in_plain_size = in_height * in_width;
+      const int32_t padding_depth = padding_list[0].first;
+      const int32_t padding_top = padding_list[1].first;
+      const int32_t padding_left = padding_list[2].first;
+      for (auto d = 0; d < out_depth; ++d)
+      {
+        for (auto h = 0; h < out_height; ++h)
+        {
+          const auto in_offset =
+            (d + padding_depth) * in_plain_size + (h + padding_top) * in_width + (padding_left);
+          const auto out_offset = (d * out_plain_size) + (h * out_width);
+          // copy a row of input data to output data
+          std::memcpy(output_data + out_offset, input_data + in_offset, out_width * sizeof(T));
+        }
+      }
+      break;
+    }
+    // NOTE: Assume that tensors' memory format is NHWC. In cker, it cannot be checked.
+    case 4:
+    {
+      const int32_t out_cube = output_shape.Dims(0);
+      const int32_t out_depth = output_shape.Dims(1);
+      const int32_t out_height = output_shape.Dims(2);
+      const int32_t out_width = output_shape.Dims(3);
+      const int32_t out_plain_size = out_height * out_width;
+      const int32_t out_cube_size = out_depth * out_plain_size;
+      const int32_t in_depth = input_shape.Dims(1);
+      const int32_t in_height = input_shape.Dims(2);
+      const int32_t in_width = input_shape.Dims(3);
+      const int32_t in_plain_size = in_height * in_width;
+      const int32_t in_cube_size = in_depth * in_plain_size;
+      const int32_t padding_cube = padding_list[0].first;
+      const int32_t padding_depth = padding_list[1].first;
+      const int32_t padding_top = padding_list[2].first;
+      const int32_t padding_left = padding_list[3].first;
+      for (auto c = 0; c < out_cube; ++c)
+      {
+        for (auto d = 0; d < out_depth; ++d)
+        {
+          for (auto h = 0; h < out_height; ++h)
+          {
+            const auto in_offset = (c + padding_cube) * in_cube_size +
+                                   (d + padding_depth) * in_plain_size +
+                                   (h + padding_top) * in_width + (padding_left);
+            const auto out_offset = (c * out_cube_size) + (d * out_plain_size) + (h * out_width);
+            // copy a row of input data to output data
+            std::memcpy(output_data + out_offset, input_data + in_offset, out_width * sizeof(T));
```

Comment:
I'm a bit afraid that I might be wrong.

But this offset calculation looks like `NCHW`. 
```
const auto in_offset = (c + padding_cube) * in_cube_size +     // n * CHW
                                   (d + padding_depth) * in_plain_size +    // c * HW 
                                   (h + padding_top) * in_width                   // h * W 
                                   + padding_left                                         // w 
``` 

I referred this : https://oneapi-src.github.io/oneDNN/dev_guide_understanding_memory_formats.html#nchw

Reply:
IHMO, in cker, the logic here doesn't matter NCHW/NHWC. In the view of Pad kernel, it just does copying. If it might be NCHW, the code var name will be changed.

If I'm wrong, I'll leave a comment.

Reply:
> IHMO, in cker, the logic here doesn't matter NCHW/NHWC.  

I agree somehow. 

IMO, layout in onert(`ir::Layout::NCHW`) just means`nnfw::cker::Shape`'s element order. 
In the tensor level, offset-calculation can be different. 

( I feel similar things while working on MaxPool cker, before )

Reply:
Okay now I'm sure somewhat.

At the train backend level, tensors' layer assume NHWC.
```c++
// backend/train/Config.cc
ir::Layout Config::supportLayout(const ir::IOperation &, ir::Layout) { return ir::Layout::NHWC; }

// backend/train/BackendContext.cc
backend::train::ITensorRegistry *BackendContext::genTrainingTensors()
{
  const ir::train::TrainableGraph &tgraph = *trainable_graph();
  auto tensor_builder = _tensor_builder;

  tgraph.operands().iterate([&](const ir::OperandIndex &ind, const ir::Operand &obj) {
    if (external_operands().contains(ind))
      return;
    // NOTE Assuming there is no layout changes (Always assume NHWC or UNKNOWN)
    assert(tgraph.layout() != ir::Layout::NCHW);

    // TODO Different shape of back propagation tensor
    ir::OperandInfo backend_info{obj.shape(), obj.typeInfo(), obj.info().memAllocType(),
                                 obj.isConstant()};
    tensor_builder->registerBackwardTensorInfo(ind, backend_info, ir::Layout::NHWC);
  });

  // TODO Plan tensor builds to reduce peak memory usage
  tgraph.operands().iterate([&](const ir::OperandIndex &ind, const ir::Operand &) {
    if (tensor_builder->isRegisteredBackward(ind))
      tensor_builder->notifyBackwardFirstUse(ind);
  });

  tensor_builder->allocateBackward();

  return _tensor_registry.get();
}
```

Therefore, at the rank-4 tensor in cker, it would be handled as NHWC.

I fixed words from specific width/height/depth to abstract(less-specific) row/col/plain. This means that CHW/HWC is not important point in cker.

Reply:
Aha, I think I also got it. 
Thank you  :+1: